course_info/static/course_info/js/controllers/DetailsController.js

```diff
@@ -72,35 +77,39 @@
         };
 
         dc.dissociateSite = function(siteListIndex) {
-            // todo: siteId is not currently available from backend; once added we can use that to call backend (instead of siteListIndex) and remove from course instance object
-            $scope.confirmDissociateSiteModalInstance = $uibModal.open({
+            dc.confirmDissociateSiteModalInstance = $uibModal.open({
                 animation: true,
                 templateUrl: 'partials/dissociate-site-confirmation.html',
-                controller: function ($scope, $uibModalInstance, siteURL) {
-                    $scope.siteURL = siteURL;
+                controller: function ($scope, $uibModalInstance, siteURL, site_map_id) {
+                    dc.siteURL = siteURL;
+                    dc.site_map_id = site_map_id;
                 },
                 resolve: {
                     siteURL: function () {
                         return dc.courseInstance.sites[siteListIndex].course_site_url;
+                    },
+                    site_map_id: function() {
+                        return dc.courseInstance.sites[siteListIndex].site_map_id;
                     }
                 }
             });
+            dc.confirmDissociateSiteModalInstance.result.then(
+                function modalConfirmed() {
+                    var delete_course_site_url = djangoUrl.reverse(dc.apiProxy,
+                        [dc.apiBase + dc.courseInstance.course_instance_id + '/sites/' + dc.courseInstance.sites[siteListIndex].site_map_id + '/']);
+                    $http.delete(delete_course_site_url)
+                        .then(function(response){
+                            dc.dissociateSiteInProgressIndex = siteListIndex;
+                            dc.courseInstance.sites.splice(siteListIndex, 1);
+                            dc.dissociateSiteInProgressIndex = null;
```

Comment:
ok, yeah, that makes sense. Will add the finally block with dc.dissociateSiteInProgressIndex = null;